lib/router/index.ts

```diff
@@ -0,0 +1,222 @@
+  return {
+    valid: res.valid,
+    errors: res.errors.map((err) => `${name}: ${err.stack}`),
+  };
+}
+
+// validateRoutes ensures that `routes` matches the config schema. We have this
+// function instead of just doing a plain jsonschema.validate in order to get
+// better error messages for the "output" object (by default jsonschema would
+// just tell you that the output block doesn't match any of the known output
+// formats, but won't tell you what's wrong because it doesn't let you
+// condition on the output.type property).
+function validateRoutes(routes) {
+  const validator = new jsonschema.Validator();
+  validator.addSchema({definitions: schemaDefs}, "/config");
+  const results = _.mapObject(routes, (rule, name) => {
+    // validate that the rule has the appropriate structure
+    const structureVal = validator.validate(rule, {
+      type: "object",
+      additionalProperties: false,
+      required: ["matchers", "output"],
+      properties: {
+        matchers: {
+          type: "object",
+        },
+        output: {
+          type: "object",
+          properties: {
+            type: {
+              type: "string",
+              enum: ["metrics", "alert", "analytics", "notification"],
+            },
+          },
+        },
+      },
+    });
+    if (!structureVal.valid) {
+      return transformValidation(structureVal, `Rule "${name}"`);
+    }
+    const matchersVal = transformValidation(
+      validator.validate(rule.matchers, {$ref: "/config#/definitions/matchers"}),
+      `Rule "${name}" matchers`
+    );
+    const outputVal = transformValidation(
+      validator.validate(rule.output, {$ref: `/config#/definitions/${rule.output.type}Output`}),
+      `Rule "${name}" output`
+    );
+    return {
+      valid: matchersVal.valid && outputVal.valid,
+      errors: matchersVal.errors.concat(outputVal.errors),
+    };
+  });
+  return _.reduce(results, (acc, res) => ({
+    valid: acc.valid && res.valid,
+    errors: acc.errors.concat(res.errors),
+  }));
+}
+
+// parseConfig parses and validates the configuration passed as a string. It
+// returns an object of the form {valid, rules, errors}, where valid is true if
+// it was successfully parsed, rules is an array of rules, and errors is an
+// array of errors.
+function parseConfig(fileString) {
+  let routes;
+  try {
+    routes = yaml.safeLoad(fileString).routes;
+  } catch (e) {
+    return {valid: false, rules: [], errors: [e]};
+  }
+  const validateRes = validateRoutes(routes);
+  if (!validateRes.valid) {
+    return _.assign(validateRes, {rules: []});
+  }
+  try {
+    const rulesObj = _.mapObject(routes, (elem, name) => new Rule(name, elem.matchers, elem.output));
+    const rules = _.values(rulesObj);
+    return {valid: true, rules, errors: []};
+  } catch (e) {
+    return {valid: false, rules: [], errors: [e]};
+  }
+}
+
+class Router {
+  rules = null;
+
+  constructor(rules) {
+    this.rules = rules || [];
```

Comment:
perhaps the constructor could accept either 

1. an array of rules (which it will validate then use)
2. a string filename (in which case it will open, YML parse the file, then continue as in (1))

Then you don't need a `loadConfig` method at all. It just happens implicitly in the constructor.